### PR TITLE
[QOL] SchematicInfoDialog Block Count Implementation

### PR DESCRIPTION
**Pull request by zethnest**
A preview of it:
Screenshot_20200710_102455

Would be cool to see the block count to see ratios are correct or not.

### CODE REVIEW

File: core/src/mindustry/ui/dialogs/SchematicsDialog.java [hunk 1]
```diff
@@ -333,6 +336,27 @@ public void show(Schematic schem){
...
+            IntIntMap blocks = new IntIntMap();
+            schem.tiles.each(t -> {
+                blocks.increment(t.block.id);
+            });
+            cont.table(t -> {
+                int i = 0;
+                for(Entry ent : blocks){
+                    int v = ent.value;
+                    t.image(Vars.content.block(ent.key).icon(Cicon.small)).left();
+                    t.label(() -> {
```
Comment (Anuken) [inline core/src/mindustry/ui/dialogs/SchematicsDialog.java:348]: This should not be a lambda label. Just use `add` with a string.

File: core/src/mindustry/ui/dialogs/SchematicsDialog.java [hunk 2]
```diff
@@ -333,6 +336,27 @@ public void show(Schematic schem){
             cont.add(new SchematicImage(schem)).maxSize(800f);
             cont.row();
 
+            IntIntMap blocks = new IntIntMap();
+            schem.tiles.each(t -> {
```
Comment (Anuken) [inline core/src/mindustry/ui/dialogs/SchematicsDialog.java:340]: This should be an expression lambda.

**Comment by XEN0PHIL** [comment]
Energy display is pointless, it will never show an accurate power consumption/generation of a complex schematic

**Comment by zethnest** [comment]
> Energy display is pointless, it will never show an accurate power consumption/generation of a complex schematic

That is from another commit. Implemented by Anuke. Please read PR title and code.

**Comment by Anuken** [comment]
How does this look for massive/diverse schematics? I'm concerned that it will run out of space on mobile landscape orientations.

**Comment by zethnest** [comment]
> How does this look for massive/diverse schematics? I'm concerned that it will run out of space on mobile landscape orientations.

It wraps after 6 blocks. I forgot to show 1 where it has a lot of it. Like this for example:
Screenshot_20200711_231716

**Comment by Anuken** [comment]
Right, but my concern is that all the wrapping will reach the bottom of the screen on landscape. It may need a scrollpane, if it doesn't have one already.

**Comment by zethnest** [comment]
Screenshot_20200712_091344
The picture above is this PR ported to 5.0 to test on mods. Unsure of the behavior of the icons, probably because of the mod. But the schematic display is shrinked. Unsure of master.

**Comment by zethnest** [comment]
I see no significant need for this. Will reopen when there are requests and further ideas for improvements.

**Pull request closed without merging**